code/go/0chain.net/smartcontract/interestpoolsc/update_test.go

```diff
@@ -0,0 +1,145 @@
+package interestpoolsc
+
+import (
+	"testing"
+	"time"
+
+	configpkg "0chain.net/chaincore/config"
+	"0chain.net/core/common"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func assertErrMsg(t *testing.T, err error, msg string) {
+	t.Helper()
+
+	if msg == "" {
+		assert.Nil(t, err)
+		return
+	}
+
+	if assert.NotNil(t, err) {
+		assert.Equal(t, msg, err.Error())
+	}
+}
+
+func getGlobalNodeTest() (gn *GlobalNode) {
+	const pfx = "smart_contracts.interestpoolsc."
+
+	configpkg.SmartContractConfig.Set(pfx+"max_mint", 100)
+	configpkg.SmartContractConfig.Set(pfx+"min_lock", 1000)
+	configpkg.SmartContractConfig.Set(pfx+"apr", 0.1)
+	configpkg.SmartContractConfig.Set(pfx+"min_lock_period", 24*time.Hour)
+
+	return &GlobalNode{
+		ADDRESS, 100 * 1e10, 0, 1000, 0.1, 24 * time.Hour,
+	}
+}
+
+func Test_getConfig(t *testing.T) {
+	var (
+		ipsc       = newTestInterestPoolSC()
+		balances   = newTestBalances()
+		configured = getGlobalNodeTest()
+		gn         = ipsc.getGlobalNode(balances, "")
+	)
+	assert.EqualValues(t, configured, gn)
+}
+
+func TestInterestPoolSmartContractUpdate(t *testing.T) {
+	var (
+		ipsc       = newTestInterestPoolSC()
+		balances   = newTestBalances()
+		tp         = common.Timestamp(0)
+		tx         = newTransaction(owner, ipsc.ID, 0, tp)
+		originalGn = getGlobalNodeTest()
+		gn         = ipsc.getGlobalNode(balances, "updateVariables")
+		err        error
+		update     = &GlobalNode{}
+	)
+
+	balances.txn = tx
+
+	// 1. Malformed update
+	t.Run("malformed update", func(t *testing.T) {
+		_, err = ipsc.updateVariables(tx, gn, []byte("} malformed {"), balances)
+		assertErrMsg(t, err, "failed to update variables: request not formatted correctly")
+	})
+
+	// 2. Non owner account tries to update
+	t.Run("non owner account", func(t *testing.T) {
+		tx.ClientID = randString(32)
+		_, err = ipsc.updateVariables(tx, gn, []byte("} malformed {"), balances)
```

Comment:
All these tests have a similar format, calling`ipsc.updateVarables` with various setup condtions. 
For consistancy can now we adopt the [table driven subtest style](https://blog.golang.org/subtests#TOC_4.) of @MurashovVen's unit tests.